tabbycat/results/views.py

```diff
@@ -295,15 +306,31 @@ def post(self, request, *args, **kwargs):
         return super().post(request, *args, **kwargs)
 
 
-class AdministratorBallotSetMixin(AdministratorMixin):
+class BallotEmailWithStatusMixin:
+    def send_email_receipts(self):
+        success = True
+        try:
+            send_email_to_adjs(DebateResult(self.ballotsub).as_dicts(), self.debate)
+        except SMTPException:
+            messages.error(self.request, _("There was a problem sending ballot receipts to adjudicators."))
+            success = False
+        except ConnectionError:
+            messages.error(self.request, _("There was a problem connecting to the e-mail server when trying to send ballot receipts to adjudicators."))
+            success = False
+
+        if success:
+            messages.success(self.request, _("Email receipts sent successfully."))
```

Comment:
We probably want to roll this into the message in `add_success_message()` method, since currently there are always two messages that show when you confirm a ballot set, that arise from the same action. The catch is that it's not obvious to me how to do this, because if the ballot set is saved but the email sending failed, you still want two messages—a success message for the saving, and an error message for the email.

I don't have any fantastic suggestions, so if you have a better way feel free to run with that, but here's one possibility:
- Have `send_email_receipts()` return `True` if it succeeded or `False` if it failed.
- `send_email_receipts()` still adds the error messages if it failed, but adds no messages if it succeeded.
- Pass the result of `send_email_receipts()` as an optional argument into `add_success_message()`, the optional argument defaulting to `False`.
- If this argument is `True`, then `add_success_message()` appends the "Email receipts sent to adjudicators." sentence to the message. (No need to check if it was confirmed here, since it should only pass in `True` if it was in the first place. Since the two sentences are grammatically independent, I'm fine with these messages being concatenated, as opposed to string-formatted.)

On another note, this structure might help simplify things a little:
``` python
try:
    send_email_to_adjs(DebateResult(self.ballotsub).as_dicts(), self.debate)
except SMTPException:
    messages.error(self.request, _("There was a problem sending ballot receipts to adjudicators."))
except ConnectionError:
    messages.error(self.request, _("There was a problem connecting to the e-mail server when trying to send ballot receipts to adjudicators."))
else:
    messages.success(self.request, _("Email receipts sent successfully."))
    # or `return True`, if implementing the message merger above
```
(If you're wondering why this pattern is used at [privateurls/views.py:219-238](https://github.com/TabbycatDebate/tabbycat/blob/develop/tabbycat/privateurls/views.py#L219-L238), but not at [privateurls/views.py:325-371](https://github.com/TabbycatDebate/tabbycat/blob/develop/tabbycat/privateurls/views.py#L325-L371), it's because the latter encompasses two `try`...`except` blocks.)